### PR TITLE
Get/set legal hold APIs

### PR DESCRIPTION
**Pull request by prakashsvmx**
Legal hold APIs support

`getObjectLegalHold(bucketName, objectName, getOpts={}, cb)`
`setObjectLegalHold(bucketName, objectName, setOpts={}, cb)`

Note:
To run all functional tests locally the below must be exported and certificates as required:
`
export MINIO_KMS_KES_ENDPOINT=https://play.min.io:7373;export MINIO_KMS_KES_KEY_FILE=root.key;export MINIO_KMS_KES_CERT_FILE=root.cert;export MINIO_KMS_KES_KEY_NAME=my-minio-key;
`

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -1401,6 +1401,89 @@ minioClient.getObjectTagging('bucketname', 'object-name', {versionId:"my-object-
...
+Get legal hold on an object.
+
+__Parameters__
+
+
+| Param  |  Type | Description  |
+|---|---|---|
+| `bucketName`  |_string_   | Name of the bucket.  |
+| `objectName`  | _string_  | Name of the object.  |
+| `setOpts`  | _object_  | Legal hold configuration options to set. e.g `{versionId:'my-version-uuid', status:'ON or OFF'}` defaults to `{}` . |
```
Comment (kanagarajkm) [inline docs/API.md:1459]: is it defaults to `ON`?

File: src/main/minio.js [hunk 2]
```diff
@@ -2829,6 +2928,8 @@ Client.prototype.setObjectLockConfig=promisify((Client.prototype.setObjectLockCo
 Client.prototype.getObjectLockConfig=promisify((Client.prototype.getObjectLockConfig))
 Client.prototype.putObjectRetention =promisify((Client.prototype.putObjectRetention))
 Client.prototype.getObjectRetention =promisify((Client.prototype.getObjectRetention))
+Client.prototype.setObjectLegalHold=promisify((Client.prototype.setObjectLegalHold))
```
Comment (kanagarajkm) [inline src/main/minio.js:2931]: extra brackets can be removed

Comment (prakashsvmx) [reply src/main/minio.js]: Thank you. I have updated.

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by kannappanr — DISMISSED** [review]
LGTM

**Review by ebozduman — CHANGES_REQUESTED** [review]

**Comment by prakashsvmx** [comment]
@ebozduman thank you for the review. I will update it.

File: src/main/minio.js [hunk 3]
```diff
@@ -2935,6 +2936,114 @@ export class Client {
...
+
+    this.makeRequest({method, bucketName, objectName, query}, '', 200, '', true, (e, response) => {
+      if (e) return cb(e)
+
+      let legalHoldConfig = Buffer.from('')
+      pipesetup(response, transformers.objectLegalHoldTransformer())
+        .on('data', data => {
+          legalHoldConfig = data
+        })
+        .on('error', cb)
```
[the remark this answers is not in the document]
Comment (prakashsvmx) [reply src/main/minio.js:2979]: In the examples, we tend to print the entire javascript error object which prints the callstack. 

I will update the examples as well to clarify this.

instead, the caller function could just use:
```
s3Client.getObjectLegalHold('bucketName', 'objectName', { versionId:'my-obj-version-uuid' }, function(err, res) {
  if (err) {
    return console.log('Unable to get legal hold config for the object', err.message) // Print only the message.
  }
  console.log(res)
})
```

**Review by kannappanr — APPROVED** [review]
LGTM

File: examples/get-object-legal-hold.js [hunk 4]
```diff
@@ -28,15 +28,24 @@ var s3Client = new Minio.Client({
...
+    return console.log('Unable to get legal hold config for the object', err.message) // Print only the message.
   }
   console.log(res)
-})
\ No newline at end of file
+})
+
+//Promise based version:
+const bucketTaggingPromise = s3Client.getObjectLegalHold('bucketName', 'objectName', { versionId:'my-obj-version-uuid' })
+bucketTaggingPromise.then((data) => {
```
Comment (ebozduman) [inline examples/get-object-legal-hold.js:46]: copy/paste issue
`bucket`~__Tagging__~`Promise` =` bucket`__Legalhold__`Promise`

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: examples/get-object-legal-hold.js (hunk 4 above)
Comment (prakashsvmx) [reply examples/get-object-legal-hold.js]: Thank you for pointing this out. I have renamed the variable.

File: docs/API.md [hunk 5]
```diff
@@ -1590,6 +1591,89 @@ minioClient.getObjectTagging('bucketname', 'object-name', {versionId:"my-object-
...
+
+| Param  |  Type | Description  |
+|---|---|---|
+| `bucketName`  |_string_   | Name of the bucket.  |
+| `objectName`  | _string_  | Name of the object.  |
+| `setOpts`  | _object_  | Legal hold configuration options to set. e.g `{versionId:'my-version-uuid', status:'ON or OFF'}` defaults to `{status:'ON'}` if not passed. |
+| `callback(err)`  | _function_  |Callback function is called with non `null` value in case of error. If no callback is passed, a `Promise` is returned.  |
+
+
+__Example__
```
Comment (ebozduman) [inline docs/API.md:1653]: This title also missed the number...

**Review by ebozduman — CHANGES_REQUESTED** [review]

**Review by ebozduman — APPROVED** [review]
LGTM when the last requested change is in.

File: docs/API.md (hunk 5 above)
Comment (prakashsvmx) [reply docs/API.md:1653]: I will update it.  :+1:  I am sorry about this.

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**